test/ebpf/testSockProg.c

```diff
@@ -0,0 +1,428 @@
+
+char *get_map_pin_path(const char *map_name)
+{
+    char *map_pin_path = malloc(sizeof(char) * (strlen(map_name) + strlen(DEFAULT_MAP_PIN_PATH_PREFIX) + 1));
+    strcpy(map_pin_path, DEFAULT_MAP_PIN_PATH_PREFIX);
+    strcat(map_pin_path, map_name);
+    return map_pin_path;
+}
+
+void get_epmap_name(int internal_map_type, int comp_id, char **full_map_name)
+{
+    printf("in get_epmap_name func\n");
+    switch (internal_map_type)
+    {
+    case COMP_POLICY_MAP:
+    {
+        int prefixlen =  strlen(COMP_PMAP_NAME_PREFIX)+5;
+        char *numVal;
+        numVal = (char*) malloc(prefixlen*sizeof(char));
+        sprintf(numVal, "%s%d",COMP_PMAP_NAME_PREFIX, comp_id);        
+        *full_map_name = numVal;
+        break;
+    }
+    case GLOBAL_POLICY_MAP:
+        *full_map_name = GLOBAL_PMAP_NAME;
+        break;
+    case IP_CACHE_MAP:
+        *full_map_name = IP_CACHE_MAP_NAME;
+        break;
+    }
+}
+
+int pin_given_map(int internal_map_type, fd_t fd)
+{
+    char *map_name = NULL;
+    get_epmap_name(internal_map_type, POLICY_MAP_ID, &map_name);
+
+    printf("pin_given_map: map name: %s\n", map_name);
+    // Map fd is invalid. Open fd to the map.
+    char *pin_path = get_map_pin_path(map_name);
+    printf("pin_given_map: map pinned path %s\n", pin_path);
+    fd_t fdnew = bpf_obj_get(pin_path);
+    if (fdnew != INVALID_MAP_FD)
+    {
+        printf("pin_given_map: found the pinned map FD\n");
+        return 0;
+    }
+
+    printf("pin_given_map: pinned map not found, creating pin for map %s\n", map_name);
+    // Map created. Now pin the map.
+    int error = bpf_obj_pin(fd, pin_path);
+    if (error != 0)
+    {
+        // close map fd.
+        _close(fd);
+        return -1;
+    }
+    return 0;
+}
+
+struct npm_endpoint_prog_t test_ebpf_prog()
+{
+    const char *connect4_program_name = "authorize_connect4";
+    const char *connect6_program_name = "authorize_connect6";
+    const char *recv4_accept_program_name = "authorize_recv_accept4";
+    const char *recv6_accept_program_name = "authorize_recv_accept6";
+    struct npm_endpoint_prog_t _npm_endpoint_prog_t;
+    struct bpf_object *object;
+    int program_fd;
+    int result = bpf_prog_load("src/endpoint_prog.o", BPF_PROG_TYPE_CGROUP_SOCK_ADDR, &object, &program_fd);
+
+    if (!object)
+    {
+        printf("object is null\n");
+        return _npm_endpoint_prog_t;
+    }
+    printf("Loaded program\n");
+    printf("%d program fd\n", program_fd);
+    printf("%d result\n", result);
+
+    if (result < 0)
+    {
+        printf("Load program failed\n");
+        return _npm_endpoint_prog_t;
+    }
+
+    _npm_endpoint_prog_t.object = object;
+
+    printf("Getting the bpf_prog for connect program\n");
+    struct bpf_program *connect_program = bpf_object__find_program_by_name(object, connect4_program_name);
+    if (!connect_program)
+    {
+        printf("%s is null\n", connect4_program_name);
+        return _npm_endpoint_prog_t;
```

Comment:
here and below through the rest of this function, this leaks object.  Should call [bpf_object__close(object)](https://microsoft.github.io/ebpf-for-windows/libbpf_8h.html#a473872ff7ab53d1ed593e7abd206f31a) when done with it.